includes/class-sensei-course.php

```diff
@@ -3095,9 +3095,9 @@ public static function output_course_enrolment_actions() {
 							esc_html__( 'Register', 'sensei-lms' ) . '</a></div>';
 					}
 				} else {
-
-					wp_register( '<div class="status register">', '</div>' );
-
+					if ( true === (bool) apply_filters( 'sensei_user_can_register_for_course', true, $post->ID ) ) {
```

Comment:
This seems like a good opportunity to clean up this `if` statement and eliminate a level of nesting.

Since we don't ever want to show the button if `sensei_user_can_register_for_course` is `false`, we could just `return` immediately in that case:
```
if ( ! (bool) apply_filters( 'sensei_user_can_register_for_course', true, $post->ID ) ) {
  return;
}
```

Now we can drop the inner nested `if` and just check `$my_courses_page_id` and `$wp_register_link`.

Reply:
Done!